app/models/targeting.rb

```diff
@@ -36,9 +36,13 @@ def resolve_hosts!
     raise ::Foreman::Exception, _('Cannot resolve hosts without a bookmark or search query') if bookmark.nil? && search_query.blank?
 
     self.search_query = bookmark.query if dynamic? && bookmark.present?
-    self.save!
-    self.touch(:resolved_at)
-    self.hosts = User.as(user.login) { Host.authorized(RESOLVE_PERMISSION, Host).search_for(search_query) }
+    self.resolved_at = Time.zone.now
+    self.validate!
+    # avoid validation of hosts objects - tey will be loaded for no reason.
```

Comment:
> Would that validation load the host record? 

Yes, presence validation loads hosts records, I saw that in the logs. Didn't debug to the actual rails method though...

> What if the validatoin was not there at all, would we still need to disable validation here?

If the validation was not set on the model, we could use any other method (although setting host_ids directly would cause an extra query).